src/v/archival/archival_metadata_stm.h

```diff
@@ -371,6 +375,13 @@ class archival_metadata_stm final : public raft::persisted_stm<> {
     cloud_storage::remote& _cloud_storage_api;
     features::feature_table& _feature_table;
     ss::abort_source _download_as;
+
+    // for observability: keep track of the number of cloud bytes "removed" by
+    // compaction. as in the size difference between the original segment(s) and
+    // compacted segment reuploaded note that this value does not correlate to
+    // the change in size in cloud storage until the original segment(s) are
+    // garbage collected.
+    size_t _compacted_replaced_bytes{0};
```

Comment:
STM looks like the wrong place for archival metrics. Maybe, instead of maintaining additional counter here we can modify `_active_operation_res` to have type `std::optional<ss::promise<result<add_segment_meta_result>>>`. Then the value could be propagated to the `ntp_archiver`. And the `ntp_archiver` will update the metric.

Reply:
it seems like the plumbing required would not be small, and the current mechanism of _active_operation_res is used by all the calls to archival_metadata_stm. I think it could be done, but it seems like a very big change just for this metric.

Another alternative is to wire the probe to the archival_metadata_stm so that it can update the counter with the result of partition_manifest::add. What do you think of this alternative? 
The counter new value would be visible before the manifest is uploaded, but eventually, they should catch up. However, there would be some misreporting when the leadership changes in the middle of the operation.